### PR TITLE
Remove didMount to stop auto focus on PasswordForm

### PR DESCRIPTION
**Pull request by Enragedsaturday**
### Details
Hidden password element was pulling focus, and accepting any text input prior to focusing on login form (email). Added a check to see if the component is visible before pulling focus on mount, added same to the login form as per maintainer request on main issue.

### Fixed Issues

$ https://github.com/Expensify/App/issues/7199

$ https://github.com/Expensify/App/issues/7133

$ https://github.com/Expensify/App/issues/7058

$ https://github.com/Expensify/App/issues/7090

### Tests

Tested on Web, iOS, and Android
1. Make sure app is on sign on page with the "phone or email" input and not password input.
2. Check that on initial load focus is on login text input. If not type characters on keyboard. (If it is pull focus and type in characters on keyboard)
3. Submit and the password component will display. 
4. Ensure the password component does not have the characters you typed in on the previous screen when focus was not on the login component.

- [ ✓] Verify that no errors appear in the JS console

### QA Steps

1. Make sure app is on sign on page with the "phone or email" input and not password input.
2. Check that on initial load focus is on login text input. If not type characters on keyboard. (If it is pull focus and type in characters on keyboard)
3. Submit and the password component will display. 
4. Ensure the password component does not have the characters you typed in on the previous screen when focus was not on the login component.

- [ ✓] Verify that no errors appear in the JS console

### Tested On

- [ ✓] Web
- [ ✓] Mobile Web
- [✓ ] Desktop
- [ ✓] iOS
- [ ✓] Android

#### Web

https://user-images.githubusercontent.com/67085884/148615986-08eb355e-b777-488b-a472-a4185e56e9e1.mp4

#### Mobile Web

https://user-images.githubusercontent.com/67085884/148616031-553e2d89-4502-49d0-a7ff-cb6b6f246ef0.mp4

#### iOS
https://user-images.githubusercontent.com/67085884/148615582-923ecb5e-df78-40e9-8d2f-cba70167ac86.mp4

#### Android

https://user-images.githubusercontent.com/67085884/148615974-742fa3a2-1a28-4756-9faf-b062e26dbe4c.mp4

#### Desktop

https://user-images.githubusercontent.com/67085884/148616718-e556c3fc-07bb-4173-9494-ace3ae3b6ba2.mov

### CODE REVIEW

**Comment by Enragedsaturday** [comment]
I have read the CLA Document and I hereby sign the CLA

**Comment by rushatgabhane** [comment]
@marcochavezf please approve the gh actions workflow, thanks!

File: src/pages/signin/PasswordForm.js
Comment (rushatgabhane) [inline src/pages/signin/PasswordForm.js:58]: Any reason not to use `!canFocusInputOnScreenFocus()` over here?
Not using it causes this jarring keyboard bug when you launch the app in the password screen.

https://user-images.githubusercontent.com/29673073/148661082-9ab4bf65-07cd-4a36-879a-aab1344a111c.mp4

**Review by rushatgabhane — COMMENTED** [review]
Looks good. Suggested some changes.

File: src/pages/signin/PasswordForm.js
Comment (rushatgabhane) [reply src/pages/signin/PasswordForm.js:58]: Ah I see, you had asked about this in the issue.
> The !canFocusInputOnScreenFocus() is only present on the login form and prevents the autofocus on touch screen devices. It seems counterintuitive to the current solution.

I still think we should keep it in both screens. Because once you open the keyboard on login screen, the keyboard stays open on password screen too. And it avoids the keyboard bug.

Thoughts?

Comment (Enragedsaturday) [reply src/pages/signin/PasswordForm.js:58]: @rushatgabhane it was never present on the password form but I agree, it should be. Not to make this more complex then it needs to be but the whole intent of the didMount function is autofocus, and  it is now being excluded from the most common use cases (all of mobile). That bug is only present on android, and could be addressed on android with platform specific code. I'll edit the current pr and wait for your feedback.

Comment (rushatgabhane) [reply src/pages/signin/PasswordForm.js]: Hmm true. Then the question is do we want to autofocus on login page for mobile? Because the current user flow is good imo.

If yes, we'll create a new issue for it.
I'm gonna start a discussion on it on slack. Feel free to join the community if you haven't already and give your input.

Comment (Enragedsaturday) [reply src/pages/signin/PasswordForm.js]: Generally agreed. I sent two emails to join slack this past week and haven't gotten an invite yet.

**Review by rushatgabhane — APPROVED** [review]
@marcochavezf PR is ready for review 😃 Tests well on all platforms.

**Comment by marcochavezf** [comment]
Thank you @Enragedsaturday for creating this PR and thanks @rushatgabhane for reviewing and testing! 😀 The PR also looks good to me, I'm going to merge it once the unit tests are fixed https://github.com/Expensify/Expensify/issues/191675.

**Comment by Enragedsaturday** [comment]
@marcochavezf Thanks. I'm unaware of how to fix the failing test "Decrypt OSbotify GPG Key". I've requested an invite to Slack twice now but haven't gotten an invite sent yet.

**Comment by rushatgabhane** [comment]
@Enragedsaturday merge main into this branch, that should fix OSBotify issue

**Comment by rushatgabhane** [comment]
> thanks @rushatgabhane for reviewing and testing! 

Cheers! 😃

**Comment by rushatgabhane** [comment]
@Enragedsaturday can you please email contributors@expensify.com again and include your Upwork profile? 
That should get you the invite.

**Review by marcochavezf — APPROVED** [review]

**Comment by kbecciv** [comment]
@Enragedsaturday The issue https://github.com/Expensify/App/issues/7199 still reproduced during the Regression. 
IMG_9432

**Comment by Enragedsaturday** [comment]
Not sure what happened there. Only miffed issue #7090.

**Comment by rushatgabhane** [comment]
@kbecciv not a regression imho. If I'm not wrong, (7199) issue was linked to this PR after this PR was merged.

**Comment by rushatgabhane** [comment]
@Enragedsaturday can you please undo changes to fixed issues? 
Screenshot_20220116-055957

**Comment by Enragedsaturday** [comment]
@rushatgabhane 

My bad. Fixed it.

**Pull request merged**
